### ISSUE TITLE
`std_instead_of_alloc` | failed to resolve: use of undeclared crate or module `alloc`

### ISSUE THREAD

**Issue by frederikhors** [rust-lang/rust-clippy#9754]
### Description

I'm trying for the first time `clippy::restriction` and I encountered: `std_instead_of_alloc`.

If I change the import from:

```rust
use std::sync::Arc;
```

to:

```rust
use alloc::sync::Arc;
```

rustc (1.64) throws with:

```log
error[E0433]: failed to resolve: use of undeclared crate or module `alloc`
  |
2 | use alloc::sync::Arc;
  |     ^^^^^ use of undeclared crate or module `alloc`
  |
  = help: add `extern crate alloc` to use the `alloc` crate
```

How to fix this?

### Version

```text
rustc 1.64.0 (a55dd71d5 2022-09-19)
binary: rustc
commit-hash: a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52
commit-date: 2022-09-19
host: x86_64-pc-windows-msvc
release: 1.64.0
LLVM version: 14.0.6
```

**Comment by Alexendoo** [comment]
The help message says what do to, in order to use `alloc` you have to do `extern crate alloc`. Unless you are creating a `no_std` crate there is no reason to use `alloc` over core though, that lint is only meant to be enabled if that is the case

**Comment by frederikhors** [comment]
Ok. But the lint should be updated by default only in that case, nope?

**Comment by Alexendoo** [comment]
The lint isn't enabled by default, you have to opt into it

**Comment by frederikhors** [comment]
You're right.
